tests/components/abode/common.py

```diff
@@ -0,0 +1,25 @@
+"""Common methods used across tests for Abode."""
+from unittest.mock import patch
+
+from homeassistant.components.abode import DOMAIN as ABODE_DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def setup_platform(hass, platform):
+    """Set up the Abode platform."""
+    mock_entry = MockConfigEntry(
+        domain=ABODE_DOMAIN,
+        data={CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"},
+    )
+    mock_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.abode.ABODE_PLATFORMS", [platform]), patch(
```

Comment:
I'm not actually sure what the purpose of this patching is for. First, the `[platform]` argument doesn't seem to match anything in the [unittest.patch docs](https://docs.python.org/3/library/unittest.mock.html#patch) and second, I'm not sure why patching the actual platforms within the integration is needed for tests. If I remove this, the test still seem to function but I copied the testing structure from the Ring platform and it's one of the areas I'm not entirely sure what its purpose is for.

Reply:
Yes, `[platform]` will go into the `new` keyword argument.